src/libcore/num/mod.rs

```diff
@@ -5250,3 +5250,73 @@ impl_from! { u32, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0"
 
 // Float -> Float
 impl_from! { f32, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0")] }
+
+// Conversion traits for non-zero integer types
+macro_rules! nzint_impl_from {
+    ($Small: ty, $Large: ty, #[$attr:meta], $doc: expr) => {
+        #[$attr]
+        #[doc = $doc]
+        impl From<$Small> for $Large {
+            /// Widens a non-zero integer without checking the value is zero.
+            ///
+            /// # Safety
```

Comment:
`Safety` section here is irrelevant, because the input type already guarantees the value is non-zero.